components/ai_chat/resources/page/state/conversation_context.tsx

```diff
@@ -570,13 +576,13 @@ export function ConversationContextProvider(props: React.PropsWithChildren) {
   }
 
   const removeImage = (index: number) => {
-    if (!context.pendingMessageImages || context.pendingMessageImages.length === 0) {
+    if (context.pendingMessageImages.length === 0) {
```

Comment:
I think this method should be simplified a bit - we don't need the different cases here:

```ts
const updatedImages = [...context.pendingMessageImages]
updatedImages.splice(index, 1)
setPartialContext({
pendingMessageImages: updatedImages
})
```

should work for all cases (including index out of range / no items).

Reply:
Fixed in https://github.com/brave/brave-core/pull/29017/commits/3b150f67df84bf03877781a396161fd2cde60ce0